language/move-prover/tests/sources/functional/opaque_native.move

```diff
@@ -4,9 +4,13 @@ module 0x42::OpaqueNative {
         pragma aborts_if_is_strict = true;
 
         // Uninterpreted spec function with axiom.
-        // Current restrictions: we cannot deal with generic functions
         fun injection(v: vector<u8>): u64;
         axiom forall v1: vector<u8>, v2: vector<u8>: v1 == v2 <==> injection(v1) == injection(v2);
+
+        // Uninterpreted generic spec function with axiom.
+        fun generic_injection<T>(x: T): u64;
```

Comment:
To specify the framework contracts, where should we put these unintrepreted functions and the axioms? In the "main(?)" hash module? Or, can there be a separate spec file to contain common spec functions and axioms?

Reply:
It depends. We could put some generic stuff together in one file; injection might be an example. In general just specific for spec file probably ok.